### PR TITLE
Zago work

### PR DESCRIPTION
**Pull request by darbymanning**

### CODE REVIEW

File: assets/popup-custom.scss.liquid [hunk 1]
```diff
@@ -0,0 +1,100 @@
+.section-popup{
```
Comment (darbymanning) [inline assets/popup-custom.scss.liquid:1]: Shopify are [deprecating native scss support](https://www.shopify.co.uk/partners/blog/deprecating-sass), so it would be better if this was a "pure" scss file which is handled by our own build pipeline (running `build:css` will compile it using rollup).

File: assets/popup-custom.scss.liquid [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  height: 100%;
+  background: #4E6B84;
+  opacity: 0.95;
+  .box-popup-custom{
+    z-index: 1;
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+    max-width: 480px;
```
Comment (darbymanning) [inline assets/popup-custom.scss.liquid:16]: We're using `rem` instead of pixel values in this project - and have a reset so that `1rem` = `10px`. So here you can get the same result with:

```suggestion
    max-width: 48rem;
```

File: assets/popup-custom.scss.liquid [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  }
+
+  .popup-image{
+    position: relative;
```
Comment (darbymanning) [inline assets/popup-custom.scss.liquid:20]: Do you need position relative here?

File: assets/popup-custom.scss.liquid [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    max-width: 480px;
```
Comment (darbymanning) [inline assets/popup-custom.scss.liquid:21]: Ditto to rem comment above. Won't repeat this for other px values!

File: assets/popup-custom.scss.liquid [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    margin: 0 auto;
+    img{
+      margin: 0 auto;
+    }
+  }
+
+  hr{
+    background-color: #54add3;
+    position: absolute;
+    width: 108.335%;
```
Comment (darbymanning) [inline assets/popup-custom.scss.liquid:31]: This doesn't seem like the right approach. I can see why you've done it as we want it to break the bounds of its container, but it's such an arbitrary value. See how this is achieved elsewhere - I think it's a negative margin approach.

File: assets/popup-custom.scss.liquid [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    position: relative;
+    max-width: 480px;
+    margin: 0 auto;
+    img{
+      margin: 0 auto;
+    }
+  }
+
+  hr{
+    background-color: #54add3;
```
Comment (darbymanning) [inline assets/popup-custom.scss.liquid:29]: Use either `var(--blue)` here or the SCSS variable. I don't mind which.

File: assets/popup-custom.scss.liquid [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    height: 277px;
+    margin: 0 auto;
+    background: #FAFAFA;
+    padding-top: 19px;
+    position: relative;
+    padding: 0 33px;
+  }
+  .popup-content-title{
+    font-family: 'nimbus-sans-extended';
+    font-style: normal;
```
Comment (darbymanning) [inline assets/popup-custom.scss.liquid:50]: You need this? what `font-style` is it inheriting?

File: assets/popup-custom.scss.liquid [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+    left: 50%;
+    top: 25px;
+    transform: translateX(-50%);
+  }
+
+  .popup-content{
+    max-width: 480px;
+    height: 277px;
+    margin: 0 auto;
+    background: #FAFAFA;
```
Comment (darbymanning) [inline assets/popup-custom.scss.liquid:43]:
```suggestion
    background: var(--exlight-gray);
```

File: assets/popup-custom.scss.liquid [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+    padding: 0 33px;
+  }
+  .popup-content-title{
+    font-family: 'nimbus-sans-extended';
+    font-style: normal;
+    font-weight: 700;
+    font-size: 36px;
+    line-height: 40px;
+    letter-spacing: -1px;
+    color: #234766;
```
Comment (darbymanning) [inline assets/popup-custom.scss.liquid:55]:
```suggestion
    color: var(--dark-blue);
```

File: assets/popup-custom.scss.liquid [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+    padding-top: 19px;
+  }
+
+  .popup-content-description{
+    font-family: 'urw-din';
+    font-style: normal;
```
Comment (darbymanning) [inline assets/popup-custom.scss.liquid:61]: Ditto comment about needing `font-style`, seems strange.

File: assets/popup-custom.scss.liquid [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+    font-weight: 400;
+    font-size: 18px;
+    line-height: 22px;
+    color: #234766;
```
Comment (darbymanning) [inline assets/popup-custom.scss.liquid:65]:
```suggestion
    color: var(--dark-blue);
```

File: assets/popup-custom.scss.liquid [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+    border: 1px solid #7D7D7D;
+    margin: 18px 0 0 0;
+  }
+
+  .bt-ct{
+    width: 100%;
+    max-width: 314px;
+    height: 31px;
+    background: #54ADD3;
```
Comment (darbymanning) [inline assets/popup-custom.scss.liquid:82]:
```suggestion
    background: var(--blue);
```

File: assets/popup-custom.scss.liquid [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+    font-style: normal;
+    font-weight: 400;
+    font-size: 18px;
+    line-height: 22px;
+    color: #234766;
+    padding-top: 5px;
+    height: 68px;
+  }
+
+  .email-ct{
```
Comment (darbymanning) [inline assets/popup-custom.scss.liquid:70]: What does `ct` mean here?

File: assets/popup-custom.scss.liquid [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+    margin: 18px 0 0 0;
+  }
+
+  .bt-ct{
+    width: 100%;
+    max-width: 314px;
+    height: 31px;
+    background: #54ADD3;
+    font-family: 'nimbus-sans-extended';
+    font-style: normal;
```
Comment (darbymanning) [inline assets/popup-custom.scss.liquid:84]: Again with `font-style`...

File: assets/popup-custom.scss.liquid [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+    font-weight: 700;
+    font-size: 18px;
+    line-height: 26px;
```
Comment (darbymanning) [inline assets/popup-custom.scss.liquid:87]: Probably better to use a times value here (`1.4`?)

File: assets/popup-custom.scss.liquid [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+    text-align: center;
+    text-transform: uppercase;
+    color: #FFFFFF;
```
Comment (darbymanning) [inline assets/popup-custom.scss.liquid:90]:
```suggestion
    color: var(--white);
```

File: layout/theme.liquid [hunk 17]
```diff
@@ -23,7 +23,8 @@
 
       {{ content_for_header }} <!-- Header hook for plugins -->
       {{ 'style.css' | asset_url | stylesheet_tag }}
-
+      {{ 'popup-custom.scss.css' | asset_url | stylesheet_tag }}
```
Comment (darbymanning) [inline layout/theme.liquid:26]: Ditto comment around deprecating SCSS, be best for this to be compiled into the existing style.css

File: layout/theme.liquid [hunk 18]
```diff
@@ -23,7 +23,8 @@
...
+      <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.0/jquery.min.js"></script>
```
Comment (darbymanning) [inline layout/theme.liquid:27]: jquery is already included i the line below, is there a reason it's being included again?

Also note we shouldn't be including scripts this way - [see Shopify's theme check guidance](https://github.com/Shopify/theme-check/blob/main/docs/checks/parser_blocking_javascript.md).

File: sections/popup-custom.liquid [hunk 19]
```diff
@@ -0,0 +1,72 @@
+<div id="section-popup" class="section-popup" style="display:none;">
+  <div class="hide-box"> </div>
```
Comment (darbymanning) [inline sections/popup-custom.liquid:2]: 1. This should probably be a button as it's more semantic and accessible.
2. It'd be good to use a `js-` namespaced class to hook into its behaviour, rather than applying JS to a "loose" class like this.

```suggestion
  <button class="hide-box js-close-pop-up"></button>
```

File: sections/popup-custom.liquid [hunk 20]
```diff
@@ -0,0 +1,72 @@
...
+  <div class="box-popup-custom">
+    <div class="popup-image">
+      {% if section.settings.image != blank %}
+      <img src="{{section.settings.image| image_url}}">
```
Comment (darbymanning) [inline sections/popup-custom.liquid:6]: Setup [theme-check](https://shopify.dev/themes/tools/theme-check) with your IDE and it will pick on somethings that need to change here. Specifically:

- Missing width and height attributes `theme-check`[ImgWidthAndHeight](https://github.com/Shopify/theme-check/blob/main/docs/checks/img_width_and_height.md)
- Add a loading="lazy" attribute to defer loading of images `theme-check`[ImgLazyLoading](https://github.com/Shopify/theme-check/blob/main/docs/checks/img_lazy_loading.md)
- Space missing after '{{' `theme-check`[SpaceInsideBraces](https://github.com/Shopify/theme-check/blob/main/docs/checks/space_inside_braces.md)

File: sections/popup-custom.liquid [hunk 21]
```diff
@@ -0,0 +1,72 @@
...
+      {% else %}
+      <img src="https://cdn.shopify.com/s/files/1/0634/1266/6600/files/placeholder-images-image_large_1.png?v=1648108064" style="width: 100%;height: 330px;">
```
Comment (darbymanning) [inline sections/popup-custom.liquid:8]: Ditto to the above, also

- Inline styles isn't good! Can we add this to the CSS please?
- An absolute URL like this doesn't seem right. Can't we use [asset_img_url](https://shopify.dev/api/liquid/filters/url-filters#asset_img_url)?

File: sections/popup-custom.liquid [hunk 22]
```diff
@@ -0,0 +1,72 @@
+<div id="section-popup" class="section-popup" style="display:none;">
```
Comment (darbymanning) [inline sections/popup-custom.liquid:1]: We're missing aria roles and labels here for screen readers. [See MDN](https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaModal#examples).

File: sections/popup-custom.liquid [hunk 23]
```diff
@@ -0,0 +1,72 @@
...
+    <div class="popup-content">
+      <div class="popup-content-title">
+        {{section.settings.header}}
+      </div>
+      <div class="popup-content-description">
+        {{section.settings.description}}
+      </div>
+
+      {%- assign formId = 'ContactFooter' -%}
+      <form action="https://manage.kmail-lists.com/subscriptions/subscribe?a=MccrpL&amp;g=H5vr9r" method="post" name="mc-embedded-subscribe-form" target="_blank">
```
Comment (darbymanning) [inline sections/popup-custom.liquid:21]: This must work I guess, but the `&amp;` seems wrong as it's a query param. I think it should be:

```suggestion
      <form action="https://manage.kmail-lists.com/subscriptions/subscribe?a=MccrpL&g=H5vr9r" method="post" name="mc-embedded-subscribe-form" target="_blank">
```

File: sections/popup-custom.liquid [hunk 24]
```diff
@@ -0,0 +1,72 @@
...
+        <input type="email" value="" name="EMAIL" class="email-ct{% if form.errors %} input--error{% endif %}" placeholder="Enter email...">
+        <button class="bt-ct" type="submit" value="Sign Up" name="subscribe">
+          <p>Subscribe</p>
```
Comment (darbymanning) [inline sections/popup-custom.liquid:24]: The `<p>` tag seems superfluous to me.

File: sections/popup-custom.liquid [hunk 25]
```diff
@@ -0,0 +1,72 @@
...
+        $(".section-popup").fadeIn(3000);
+    $("body").css("overflow", "hidden");
+        localStorage.setItem('popState','shown')
+    }
+
+  $('.hide-box').click(function(){
+    $(".section-popup").fadeOut();
+    $("body").css("overflow", "auto");
+  });
+});
```
Comment (darbymanning) [inline sections/popup-custom.liquid:42]: We really ought to abstract this into a JS file and include it in `application.js`

File: sections/popup-custom.liquid [hunk 26]
```diff
@@ -0,0 +1,72 @@
...
+      }
+	]
+  }
+{% endschema %}
+
+{% stylesheet %}
+{% endstylesheet %}
+
+{% javascript %}
+{% endjavascript %}
```
Comment (darbymanning) [inline sections/popup-custom.liquid:72]:
```suggestion
```
